src/win32/tip/build_tip_forwarder_dll.py

```diff
@@ -0,0 +1,354 @@
+#     * Redistributions in binary form must reproduce the above
+# copyright notice, this list of conditions and the following disclaimer
+# in the documentation and/or other materials provided with the
+# distribution.
+#     * Neither the name of Google Inc. nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+"""This script builds am Arm64X pure forwarder DLL for Mozc TIPs on Windows.
+
+See the following document for more details:
+https://learn.microsoft.com/en-us/windows/arm/arm64x-build#building-an-arm64x-pure-forwarder-dll
+"""
+
+
+import argparse
+import dataclasses
+import os
+import pathlib
+import shutil
+import subprocess
+import tempfile
+from typing import Union
+
+from build_tools import mozc_version
+from build_tools import vs_util
+
+
+@dataclasses.dataclass
+class ForwarderInfo:
+  """Information about the forwarder DLL to be generated."""
+
+  branding: str
+  version: mozc_version.MozcVersion
+
+  def get_version_string(self, separator='.') -> str:
+    components = ['@MAJOR@', '@MINOR@', '@BUILD@', '@REVISION@']
+    return self.version.GetVersionInFormat(separator.join(components))
+
+  @property
+  def forwarder_dll_name(self) -> str:
+    return {
+        'Mozc': 'mozc_tip64x.dll',
+        'GoogleJapaneseInput': 'GoogleIMEJaTIP64X.dll',
+    }[self.branding]
+
+  @property
+  def x64_impl_dll_name(self) -> str:
+    return {
+        'Mozc': 'mozc_tip64.dll',
+        'GoogleJapaneseInput': 'GoogleIMEJaTIP64.dll',
+    }[self.branding]
+
+  @property
+  def arm64_impl_dll_name(self) -> str:
+    return {
+        'Mozc': 'mozc_tip64arm.dll',
+        'GoogleJapaneseInput': 'GoogleIMEJaTIP64Arm.dll',
+    }[self.branding]
+
+  @property
+  def file_description(self) -> str:
+    return {
+        'Mozc': 'Mozc TIP Module Forwarder',
+        'GoogleJapaneseInput': 'Google 日本語入力 TIP モジュール フォワーダー',
+    }[self.branding]
+
+  @property
+  def product_name(self) -> str:
+    return {
+        'Mozc': 'Mozc',
+        'GoogleJapaneseInput': 'Google 日本語入力',
+    }[self.branding]
+
+  @classmethod
+  def _get_def_file_content(filename: str) -> str:
```

Comment:
Please add `cls` as the first argument.

```
def _get_def_file_content(cls, filename: str) -> str:
```

Reply:
After re-reading [the relevant section](https://google.github.io/styleguide/pyguide.html#217-function-and-method-decorators) in the style guide, it looks like a module-level function looks to be a better solution in this case. So extracted it as a module-level function.

> Never use `staticmethod` unless forced to in order to integrate with an API defined in an existing library. Write a module-level function instead.